web/modules/custom/joinup_group/src/JoinupGroupRelationInfoInterface.php

```diff
@@ -2,17 +2,17 @@
 
 declare(strict_types = 1);
 
-namespace Drupal\joinup_core;
+namespace Drupal\joinup_group;
 
 use Drupal\Core\Entity\EntityInterface;
 use Drupal\Core\Session\AccountInterface;
 use Drupal\og\OgMembershipInterface;
 use Drupal\rdf_entity\RdfInterface;
 
 /**
- * An interface for Joinup relation manager services.
+ * An interface for services that provide information about group relations.
  */
-interface JoinupRelationManagerInterface {
+interface JoinupGroupRelationInfoInterface {
 
   /**
    * Retrieves the parent of the entity.
```

Comment:
This method is useless and should be removed. Instead of:

```php
/** @var \Drupal\joinup_group\JoinupGroupRelationInfoInterface $relation_info */
$relation_info = \Drupal::service('joinup_group.relation_info');
$parent = $parent = $relation_info->getParent($node);
```

we should use:

```php
$parent = $node->get('og_audience')->entity;
```

or:

```php
$parent = $node->get(OgGroupAudienceHelperInterface::DEFAULT_FIELD)->entity;
```

Reply:
I looked into this and it seems this method is intended to be used in places where we deal with any type of node or even other entities. I see that in most cases where multiple entity types are supported it is first filtering on nodes though for efficiency.

For example if any code that deals with community content and other node types cannot use this because only group related content has the `og_audience` field. We cannot guarantee that every node type we will ever support in the future will have this field, or even that it will have this machine name.

For the code that deals exclusively with community content we can remove it though.

Reply:
> For example if any code that deals with community content and other node types cannot use this because only group related content has the `og_audience` field. We cannot guarantee that every node type we will ever support in the future will have this field, or even that it will have this machine name.

Yes, but we have no use cases. So, if we remove the actual usages, it's dead code. I wonder if we should build/keep an unused method just for a hypothetical, unlikely future usage.

Reply:
Hmmm looking deeper it seems that if this method is called on non-group entities then the current implementation will throw an exception inside `MembershipManager::getGroupIds()` so my assumption is wrong, this method _cannot_ be called on non-group content.